src/sleep/sleep.c

```diff
@@ -63,16 +64,14 @@ static int write_efi_hibernate_location(const HibernateLocation *hibernate_locat
         const char *uuid_str;
         sd_id128_t uuid;
         struct utsname uts = {};
-        int log_level, log_level_ignore;
+        int r;
 
         assert(hibernate_location);
         assert(hibernate_location->swap);
 
         if (!is_efi_boot())
-                return 0;
-
-        log_level = required ? LOG_ERR : LOG_DEBUG;
-        log_level_ignore = required ? LOG_WARNING : LOG_DEBUG;
+                return log_full_errno(log_level, SYNTHETIC_ERRNO(EOPNOTSUPP),
+                                      "Not an EFI boot, passing HibernateLocation not possible.");
```

Comment:
maybe say more explicitly: `passing HibernateLocation via EFI variable is not not possible, refusing hibernation.`